src/bluealsa.h

```diff
@@ -49,6 +49,10 @@ struct ba_config {
 	 * during profile registration. Leave it empty to use any adapter. */
 	GArray *hci_filter;
 
+	/* Device connection sequence number */
+	pthread_mutex_t seq_mutex;
+	uint32_t seq;
```

Comment:
Some time ago I thought about using atomics in this project. It would require C11 as a minimum compiler, though. However, I'm not sure whether bluez-alsa can be built with pure C99, so maybe the requirement for C11 is already there. Currently, there is no explicit atomic usage, because there is no real use case (ref counts for ba_adapter, ba_device and ba_transport will need mutex anyway). But this counter seems like a legit atomic :) What do you think?

Reply:
OK, I've now used atomics for this.